includes/class-wp-job-manager-rest-api.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_REST_API.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handles functionality related to the REST API.
+ *
+ * @since 1.33.0
+ */
+class WP_Job_Manager_REST_API {
+	/**
+	 * Sets up initial hooks.
+	 *
+	 * @static
+	 */
+	public static function init() {
+		add_action( 'rest_prepare_job_listing', array( __CLASS__, 'prepare_job_listing' ), 10, 2 );
+	}
+
+	/**
+	 * Filters the job listing data for a REST API response.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WP_Post          $post     Post object.
+	 * @return WP_REST_Response
+	 */
+	public static function prepare_job_listing( $response, $post ) {
```

Comment:
Is this function just hiding the field from being returned? I'm surprised that there isn't a way to add this as a callback to `register_meta`. I wonder if it would make more sense to call the `auth_view_callback` function when we call `register_meta` and use its value to determine `show_in_rest`? So it might look like:

```php
'show_in_rest' => $field['show_in_rest'] && call_user_func( $field['auth_view_callback'], ... ),
```

Maybe I'm bikeshedding here, and if so don't worry about it. It just seems like there should be a better way to do this than to add the field to the API response and then use a filter to remove it?

Reply:
Sorry, I was going to add a comment and forgot. The reason I did this filtering here instead of when deciding whether or not to `show_in_rest` is because for some auth checks we need `$post` in order to match existing auth check logic (`edit_others_posts` or `job_manager_user_can_edit_job( $post_id )`), which might not be available by the time we register our meta fields. I also expected there to be something for this in `register_meta`, but this was the best I could find.